devito/data.py

```diff
@@ -273,18 +358,18 @@ class Data(np.ndarray):
 
     def __new__(cls, shape, dimensions, dtype, allocator=ALLOC_FLAT):
         assert len(shape) == len(dimensions)
-        ndarray, c_pointer = allocator.alloc(shape, dtype)
+        ndarray, free_args = allocator.alloc(shape, dtype)
         obj = np.asarray(ndarray).view(cls)
         obj._allocator = allocator
-        obj._c_pointer = c_pointer
+        obj._free_args = free_args
```

Comment:
I'd call this ``_memfree_args`` (or whatever, as long as it's got a reference to "memory"), because ``_free_args`` may be misinterpreted